cvat/apps/iam/urls.py

```diff
@@ -35,15 +35,41 @@
 if settings.IAM_TYPE == "BASIC":
     urlpatterns += [
         path("register", RegisterViewEx.as_view(), name=BASIC_REGISTER_PATH_NAME),
+    ]
+
+    password_change_view = PasswordChangeView
+
+    if "cvat.apps.api_tokens" in settings.INSTALLED_APPS:
+        import importlib
+
+        from rest_framework.decorators import authentication_classes
+
+        from cvat.apps.api_tokens.authentication import ApiTokenAuthentication
+
+        no_api_token_auth_classes = []
+        for auth_class_path in settings.REST_FRAMEWORK["DEFAULT_AUTHENTICATION_CLASSES"]:
+            auth_class_module_path, auth_class_name = auth_class_path.rsplit(".", maxsplit=1)
+            auth_class_module = importlib.import_module(auth_class_module_path)
+            auth_class = getattr(auth_class_module, auth_class_name)
+
+            if not issubclass(auth_class, ApiTokenAuthentication):
+                no_api_token_auth_classes.append(auth_class)
+
+        password_change_view = authentication_classes(no_api_token_auth_classes)(
+            password_change_view,
+        )
```

Comment:
Seems like it'd be simpler to do `PasswordChangeView.authentication_classes = no_api_token_auth_classes`. That's all that the decorator does.

Reply:
Using a decorator seems to be the designed way to do this in Django.

Reply:
Wait a minute... there is an even better way to do this - just pass `authentication_classes` when calling `as_view`. This avoids modifying the original class.